viadot/sources/genesys.py

```diff
@@ -0,0 +1,526 @@
+            final_file_name = f"{output_file_name}.{file_extension}"
+
+        response_data = response_file.content
+        df = pd.read_csv(StringIO(response_file.content.decode("utf-8")))
+        if drop_duplicates is True:
+            df.drop_duplicates(inplace=True, ignore_index=True)
+        df.to_csv(f"{path}{final_file_name}", index=False, sep=sep)
+
+    def download_all_reporting_exports(
+        self,
+        store_file_names: bool = True,
+    ) -> List[str]:
+        """
+        Get information form data report and download all files.
+
+        Args:
+            g_instance (Genesys): instance of Genesys source
+            ids_mapping (Dict[str, Any], optional): relationship between id and file name. Defaults to None.
+            data_report (List[str], optional): data extracted from genesys. Defaults to None.
+            file_extension (Literal[xls, xlsx, csv;], optional): file extensions for downloaded files. Defaults to "csv".
+            store_file_names (bool, optional): decide whether to store list of names.
+        Returns:
+            List[str]: all file names of downloaded files
+        """
+        file_name_list = []
+        temp_ids_mapping = self.ids_mapping
+        if temp_ids_mapping is None:
+            self.logger.warning("IDS_MAPPING is not provided in you credentials.")
+        else:
+            self.logger.info("IDS_MAPPING loaded from local credential.")
+
+        for single_report in self.report_data:
+            file_name = (
+                temp_ids_mapping.get(single_report[2]) + "_" + single_report[-1]
+            ).upper()
+            self.download_report(
+                report_url=single_report[1],
+                output_file_name=file_name,
+                file_extension=self.file_extension,
+            )
+            if store_file_names is True:
+                file_name_list.append(file_name + "." + self.file_extension)
+
+        self.logger.info("Al reports were successfully dowonload.")
+
+        if store_file_names is True:
+            self.logger.info("Successfully genetared file names list.")
+            return file_name_list
+
+    def get_analitics_url_report(self):
+        """Fetching analytics report url from json response.
+
+        Returns:
+            string: url for analytics report
+        """
+        response = handle_api_response(
+            url=f"https://api.{self.environment}/api/v2/analytics/reporting/schedules/{self.schedule_id}",
+            headers=self.authorization_token,
+        )
+        try:
+            response_json = response.json()
+            report_url = response_json.get("lastRun", None).get("reportUrl", None)
+            self.logger.info("Successfully downloaded report from genesys api")
+            return report_url
+        except AttributeError as e:
+            self.logger.error(
+                "Output data error: " + str(type(e).__name__) + ": " + str(e)
+            )
+
+    def get_all_schedules_job(self):
```

Comment:
If you are returning `json` you should add `-> json:`.

Reply:
line 394 `return response_json` in try / except block.